File: 01-sorting/05-selection-sort.py
def selection_sort(arr):
    for i in range(len(arr)):
        smallest_idx = i
        for j in range(i + 1, len(arr)):
            if arr[j] < arr[smallest_idx]:
                smallest_idx = j
        arr[i], arr[smallest_idx] = arr[smallest_idx], arr[i]
    return arr

arr = [4, 2, 7, 3, 6, 2, 5, 1]
print("Sorted array is:", selection_sort(arr))

File: 01-sorting/go/05-selection-sort.go
package main

func selectionSort(arr []int) {
	n := len(arr)
	for i := 0; i < n-1; i++ {
		minIdx := i
		for j := i + 1; j < n; j++ {
			if arr[j] < arr[minIdx] {
				minIdx = j
			}
		}
		// Swap the found minimum element with the first unsorted element
		arr[i], arr[minIdx] = arr[minIdx], arr[i]
	}
}
